### PR TITLE
added help command

### PR DESCRIPTION
**Pull request by polyen**

### CODE REVIEW

File: cli_bot.py [hunk 1]
```diff
@@ -54,9 +55,34 @@ def __init__(self, address_book: AddressBook):
...
+        command_output = ""
+        if len(args) > 1:
+            command_output += (
+                "Warning: Command optionally expect one argument - command name. "
+                f"Received: {' '.join(args)}\n"
+            )
+
+        search_command = args[0] if args else None
+
+        if search_command:
```
Comment (TrickyOldFox) [inline cli_bot.py:73]: An early return might be a bit more readable option here. Something like:
```
if search_command:
    if not search_command in self.supported_commands:
        return "Not supported"
    
    return "Help for specific command"

return "Generic help message"
```

File: cli_bot.py [hunk 2]
```diff
@@ -282,17 +308,17 @@ def get_contact(self, *args: str) -> str:
...
         Returns:
             Command output.
         """
         command_output = ""
         if len(args) != 1:
-            raise CommandOperationalError (
+            raise CommandOperationalError(
                 "Warning: Command expect one argument: number of days. "
                 f"Received: {' '.join(args)}\n"
             )
```
Comment (TrickyOldFox) [inline cli_bot.py:329]: Can we add some more fine tuned message for cases when user provided not an integer as a number of days ? f. ex. :

```
try:
     days = int(args[0])
except ValueError:
     raise CommandOperationalError("Command expects a valid integer - number of days, please recheck your input.")

File: cli_bot.py [hunk 3]
```diff
@@ -54,9 +55,34 @@ def __init__(self, address_book: AddressBook):
...
+                "Warning: Command optionally expect one argument - command name. "
+                f"Received: {' '.join(args)}\n"
+            )
+
+        search_command = args[0] if args else None
+
+        if search_command:
+            if search_command in self.supported_commands:
+                command_output += f"Command '{search_command}' help: "
+                command_output += f"\n{self.supported_commands[search_command].__doc__.split('Raises:')[0].strip()}"
```
Comment (TrickyOldFox) [inline cli_bot.py]: This looks a bit dangerous, there can be commands that doesn't raise anything (For example print_all, it has warnings only). Is there a better way?

**Review by TrickyOldFox — COMMENTED** [review]
LGTM, but can be improved

File: cli_bot.py (hunk 3 above)
Comment (polyen) [reply cli_bot.py]: Does not split return a full string in this case?

**Review by TrickyOldFox — APPROVED** [review]

File: cli_bot.py (hunk 3 above)
Comment (TrickyOldFox) [reply cli_bot.py]: Ack

Comment (TrickyOldFox) [reply cli_bot.py]: Btw, can we split by 'Returns:' instead ? It doesn't seem like it's a meaningful info for a user to know that handler function will return a Command output

Comment (polyen) [reply cli_bot.py]: make sense

**Pull request merged**